bookwyrm/models/readthrough.py

```diff
@@ -24,3 +31,26 @@ def save(self, *args, **kwargs):
         self.user.last_active_date = timezone.now()
         self.user.save()
         super().save(*args, **kwargs)
+
+    def create_update(self):
+        if self.progress:
+            return self.progressupdate_set.create(
+                user=self.user,
+                progress=self.progress,
+                mode=self.progress_mode)
+
+class ProgressUpdate(BookWyrmModel):
+    ''' Store progress through a book in the database. '''
+    user = models.ForeignKey('User', on_delete=models.PROTECT)
+    readthrough = models.ForeignKey('ReadThrough', on_delete=models.PROTECT)
```

Comment:
When I try to delete a readthrough that has an associated progress update, I get an error because of the protected relationship. I think this is a case where CASCADE would be better than PROTECT?

Reply:
Good call, if a user is deleting a readthrough there's no reason to keep the progress updates. There's already a confirmation dialog for readthroughs, I bet I can add a "...and its X progress updates" to that just to make it clear.